src/zenml/integrations/label_studio/annotators/label_studio_annotator.py

```diff
@@ -0,0 +1,661 @@
+        if config.storage_type == "azure":
+            storage_sources = self._get_azure_import_storage_sources(dataset_id)
+        elif config.storage_type == "gcs":
+            storage_sources = self._get_gcs_import_storage_sources(dataset_id)
+        elif config.storage_type == "s3":
+            storage_sources = self._get_s3_import_storage_sources(dataset_id)
+        else:
+            raise NotImplementedError(
+                f"Storage type '{config.storage_type}' not implemented."
+            )
+        return any(
+            (
+                source.get("presign") == config.presign
+                and source.get("bucket") == uri
+                and source.get("regex_filter") == config.regex_filter
+                and source.get("use_blob_urls") == config.use_blob_urls
+                and source.get("title") == dataset.get_params()["title"]
+                and source.get("description") == config.description
+                and source.get("presign_ttl") == config.presign_ttl
+                and source.get("project") == dataset_id
+            )
+            for source in storage_sources
+        )
+
+    def get_parsed_label_config(self, dataset_id: int) -> Dict[str, Any]:
+        """Returns the parsed Label Studio label config for a dataset.
+
+        Args:
+            dataset_id: Id of the dataset.
+
+        Returns:
+            A dictionary containing the parsed label config.
+
+        Raises:
+            ValueError: If no dataset is found for the given id.
+        """
+        # TODO: check if client actually is connected etc
+        ls = self._get_client()
+        dataset = ls.get_project(dataset_id)
+        if dataset:
+            return cast(Dict[str, Any], dataset.parsed_label_config)
+        else:
+            raise ValueError("No dataset found for the given id.")
+
+    def connect_and_sync_external_storage(
+        self,
+        uri: str,
+        config: LabelStudioDatasetSyncConfig,
+        dataset: Project,
+    ) -> Optional[Dict[str, Any]]:
+        """Syncs the external storage for the given project.
+
+        Args:
+            uri: URI of the storage source.
+            config: Configuration for the dataset.
+            dataset: Label Studio dataset.
+
+        Returns:
+            A dictionary containing the sync result.
+
+        Raises:
+            ValueError: If the storage type is not supported.
+        """
+        if self._storage_source_already_exists(uri, config, dataset):
+            return None
+        if config.storage_type == "azure":
+            if not config.azure_account_name or not config.azure_account_key:
+                logger.warn(
+                    "Authentication credentials for Azure aren't fully "
+                    "provided. Please update the storage synchronization "
+                    "settings in the Label Studio web UI as per your needs."
+                )
+            storage = dataset.connect_azure_import_storage(
```

Comment:
I think you could reduce a lot of duplicated code here:
```python
args  = {
   prefix=config.prefix,
   ...  # add all the identical args here
}
if config.storage_type == "azure":
    storage = dataset.connect_azure_import_storage(
        container=uri,
        ...,  # add all the azure custom args here
        **args
    )
```

Reply:
Good point. Reimplemented this.

Reply:
This is a cool trick! TIL.